cloud/services/storage/keypair.go

```diff
@@ -0,0 +1,98 @@
+package storage
+
+import (
+	"errors"
+	"fmt"
+
+	infrastructurev1beta1 "github.com/outscale-dev/cluster-api-provider-outscale.git/api/v1beta1"
+	tag "github.com/outscale-dev/cluster-api-provider-outscale.git/cloud/tag"
+	osc "github.com/outscale/osc-sdk-go/v2"
+)
+
+type OscKeyPairInterface interface {
+	CreateKeyPair(spec *infrastructurev1beta1.OscKeypair, keypairName string) (*osc.KeypairCreated, error)
+	DeleteKeyPair(keypairName string) error
+	ReadKeyPair(keypairName string, keypairFingerprints string) (*osc.Volume, error)
+}
+
+func (s *Service) CreateKeyPair(spec *infrastructurev1beta1.OscKeypair, keypairName string) (*osc.KeypairCreated, error) {
+	keyPairName := spec.Name
+	publicKey := spec.PublicKey
+
+	keyPairRequest := osc.CreateKeypairRequest{
+		KeypairName: keyPairName,
+		PublicKey:   &publicKey,
+	}
+
+	oscApiClient := s.scope.GetApi()
+	oscAuthClient := s.scope.GetAuth()
+
+	keyPairResponse, httpRes, err := oscApiClient.KeypairApi.CreateKeypair(oscAuthClient).CreateKeypairRequest(keyPairRequest).Execute()
+	if err != nil {
+		fmt.Printf("Error with http result %s", httpRes.Status)
+		return nil, err
+	}
+
+	var resp osc.ReadKeypairsResponse
+
+	if len(resp.GetKeypairs()) < 1 {
+		return nil, fmt.Errorf("Unable to find key pair, please provide a better query criteria ")
```

Comment:
You check this before, right ?

Return a message api can not found your new keypair (something like that)